ruby_on_rails/bootstrap.md

```diff
@@ -30,3 +36,52 @@ config.label_class = 'form-control-label'
 config.input_class = 'form-control'
 ```
 
+To make the error highlighting work you need to add some css to your application
+
+```scss
+.invalid-feedback {
+  display: block;
+}
+
+.has-danger {
+  .form-control {
+    border-color: $form-feedback-invalid-color;
+  }
+}
+```
+
+For the styling of the pull down date selectors or checkboxes, you need to write some wrappers, that you can add
+to the input element.
+It is best to create a separate config file for this.
+
+```ruby
+# config/initializers/simple_form_bootstrap.rb
+
+SimpleForm.setup do |config|
+  config.wrappers :inline_date, tag: 'div', error_class: 'has-danger' do |b|
+    b.use :html5
+    b.use :label, class: 'control-label'
+    b.wrapper tag: 'div', class: 'form-inline row' do |ba|
+      ba.use :input, class: 'form-control form-inline', wrap_with: { class: 'col-md-6' }
+      ba.use :error, wrap_with: { tag: 'span', class: 'invalid-feedback' }
+      ba.use :hint, wrap_with:  { tag: 'p', class: 'help-block' }
+    end
+  end
+
+  config.wrappers :inline_checkbox, tag: 'div', class: 'control-group', error_class: 'has-error' do |b|
+    b.use :html5
+    b.wrapper tag: 'div', class: 'controls' do |ba|
+      ba.use :label_input, wrap_with: { class: 'checkbox inline' }
+      ba.use :error, wrap_with: { tag: 'span', class: 'help-inline' }
+      ba.use :hint, wrap_with: { tag: 'p', class: 'help-block' }
+    end
+  end
+end
+
+```
+
+Then you can add one to the input element:
+
+```ruby
+f.input :check, wrapper: :inline_checkbox
```

Comment:
you could also configure them as default wrapper so you don't have to specify it everytime, right?

Reply:
I didn't manage find out if that's possible.

Reply:
`config.wrapper_mappings = { ... }`